test/jest/__mock__/reactFinalFormListeners.mock.js

```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+jest.mock('react-final-form-listeners', () => {
+  return {
+    ...jest.requireActual('react-final-form-listeners'),
+
+    OnChange: ({ children }) => <>{children()}</>,
```

Comment:
You can mock the return value with `jest.fn` so lint doesn't whine about prop-types:
```
    OnChange: jest.fn(({ children }) => <>{children()}</>),
```